platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/Browser.java

```diff
@@ -199,8 +200,43 @@ private Browser(final Set<Integer> localNodesToStart) throws IOException {
                 .withConfigDataType(MetricsConfig.class)
                 .withConfigDataType(PrometheusConfig.class)
                 .withConfigDataType(OSHealthCheckConfig.class)
-                .withConfigDataType(WiringConfig.class)
-                .build();
+                .withConfigDataType(WiringConfig.class);
+
+        // The following variables are not final while being initialized.
+        ApplicationDefinition appDefinitionInit = null;
+        Map<Long, SwirldMain> appMainsInit = null;
+        Configuration config = null;
+        Exception exception = null;
+        try {
+            // Load config.txt file, parse application jar file name, main class name, address book, and parameters
+            appDefinitionInit = ApplicationDefinitionLoader.load(configurationProperties, localNodesToStart);
+
+            // Load all SwirldMain instances for locally run nodes.
+            appMainsInit = loadSwirldMains(appDefinitionInit, localNodesToStart);
+
+            // We assume all locally run instances would provide the same config to the configuration builder.
```

Comment:
@cody-littley does that make sense? Why not iterating over all?

Reply:
@hendrikebbers Does it cause problems if we register the same type with the configuration builder more than once?

- if "no", then there is no problem iterating over all platforms and calling `updateConfigurationBuilder()`, although I can't imagine any workflows we currently have where different nodes will register different configurations
- if "yes", we will need to put up with this hack until we stop accessing the configuration object with static getters (in that scenario, each node will have its own config instance)

Reply:
Tracing through the code, adding a config class puts the class into a queue.  The queue is processed in initialization by loading an instance of each class and then adding it to the configDataCache, which is a map.   If all the instances are the same record, there is no harm is doing this.  Each instance replaces the last instance added.   It's just wasted CPU cycles in loading the class multiple times.